action/protocol/staking/protocol.go

```diff
@@ -338,7 +339,9 @@ func (p *Protocol) handle(ctx context.Context, act action.Action, csm CandidateS
 	}
 
 	if receiptErr, ok := err.(ReceiptError); ok {
-		log.L().Debug("Non-critical error when processing staking action", zap.Error(err))
+		actionCtx := protocol.MustGetActionCtx(ctx)
```

Comment:
this could panic

Reply:
ctx is passed from above with action context
MustGetActionCtx(ctx) is also called in downstream funcs like handleCreateStake(), it should not panic